packages/shared/src/cache.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Stores values with an expiration time at time of insertion.
+ * Does not update the expiration time on retrieval.
+ * Values are automatically removed from the cache after the TTL expires.
+ * The cache is cleaned up periodically based on the TTL so it does
+ * not grow indefinitely.
+ */
+export class TimedCache<T> {
+  readonly #cache: Map<string, {value: T; expiresAt: number}>;
+  readonly #ttlMs: number;
+  readonly #intervalHandle: NodeJS.Timeout;
+
+  constructor(ttlMs: number) {
+    this.#cache = new Map();
+    this.#ttlMs = ttlMs;
+    this.#intervalHandle = setInterval(() => {
```

Comment:
Why are we doing this at an interval? We know when there is an entry to expire and we know when that entry should be collected.

You can use a single setTimeout to the smallest time.

For zero-cache I ended up overshooting slightly to reduce DB churn but I'm not sure that is a better design here.